src/core/common/settings.cpp

```diff
@@ -154,8 +155,9 @@ otError Settings::SaveNetworkInfo(const NetworkInfo &aNetworkInfo)
 {
     otError     error = OT_ERROR_NONE;
     NetworkInfo prevNetworkInfo;
+    uint16_t    length;
```

Comment:
```suggestion
    uint16_t    length = sizeof(NetworkInfo);
```
I think we need to init `length` since it would be used by `Read()`.